sdk/samples/core/mosquitto_rpc_client_sample.c

```diff
@@ -247,55 +289,27 @@ int main(int argc, char* argv[])
   az_result rc;
 
   // infinite execution loop
-  for (int i = 45; !sample_finished && i > 0; i++)
+  for (int i = 0; !sample_finished; i++)
   {
-    // remove any expired commands
-    pending_command* expired_command = get_first_expired_command(pending_commands);
-    while (expired_command != NULL)
-    {
-      printf(LOG_APP_ERROR "command ");
-      print_correlation_id(expired_command->correlation_id);
-      printf(" expired\n");
-      az_result ret = remove_command(&pending_commands, expired_command->correlation_id);
-      if (ret != AZ_OK)
-      {
-        printf(LOG_APP_ERROR "Expired command not a pending command\n");
-      }
-      expired_command = get_first_expired_command(pending_commands);
-    }
-#ifdef _WIN32
-    Sleep((DWORD)1000);
-#else
-    sleep(1);
-#endif
     printf(LOG_APP "Waiting...\r");
     fflush(stdout);
+
+    // remove any expired commands from the client
+    remove_expired_commands();
```

Comment:
There is a race condition if the MQTT stack operates on a different thread (e.g. Mosquitto, AsyncPaho).
The `pending_commands` object is now a shared resource and it must be guarded by a mutex.

Reply:
I have plans to move the pending_commands functionality within the sdk, where we could use the platform mutex. I'm still figuring out the level of data that might be needed for different solutions since it won't be customer modifiable if it's in the sdk. This isn't a new issue in this PR, so I'll address this in a follow-up PR either moving the pending commands functionality to the sdk or implementing mutexes in the samples.

Reply:
Added mutexes to the server sample pending_command struct and the client_pending_commands file